pandas/core/nanops.py

```diff
@@ -460,6 +460,14 @@ def nanmean(values, axis=None, skipna=True, mask=None):
     elif is_float_dtype(dtype):
         dtype_sum = dtype
         dtype_count = dtype
+    elif is_datetime64_dtype(dtype) or is_datetime64tz_dtype(dtype):
+        from pandas import DatetimeIndex
```

Comment:
Why does this need to be boxed in an index? Shouldn't `values` be a DatetimeArray right now? Or is it not yet since we haven't implemented DTA as an extension array?

Reply:
I don't know this module especially well, but my assumption is that it could be a numpy array at this point.

Reply:
I suppose the exception will be `Series[datetime64[ns]]`, which may pass an ndarray here...